cai_causal_graph/causal_graph.py

```diff
@@ -2104,7 +2104,7 @@ def from_gml_string(cls, gml: str) -> CausalGraph:
 
     @classmethod
     def from_adjacency_matrix(
-        cls, adjacency: numpy.ndarray, node_names: Optional[List[Union[NodeLike, int]]] = None
+        cls, adjacency: numpy.ndarray, node_names: Optional[List[Union[NodeLike, int]]] = None, validate: bool = False
```

Comment:
@steven-causalens should we add `validate` flag to other `from_` methods? For example, `from_networkx` calls `from_adjacency_matrix` so we can just pipe the `validate` through. Probably for consistency it is good to include in all `from_`.